### ISSUE TITLE
Can't access the Web UI

### ISSUE THREAD

**Issue by johnrix** [Jsakkos/mkv-episode-matcher#81]
I am trying to get this working in order to organise my rips of Friends for my wife who is missing them after having been removed from Netflix!

I have tried installing this on Windows and on Ubuntu in WSL2, but no matter which way I install it and launch it, I simply cannot get anything out of the web UI (localhost:8001) besides:

{"detail":"Not Found"}

I am at least able to access the /docs path, and trying out some of the endpoints is working (I can update config, list files in the media folder), so there is that, but I would love to be able to see what the tool can actually do. What am I doing wrong? Thanks!

I did find a thread of yours on Reddit where you mentioned trying to configure it with "mkv-match --get-subs" (I was not prompted for any credentials when first launching it on either platform). This command has no notable effect though and still just launches the server.

For reference, here is what I am seeing when launching it from PowerShell (admittedly, I am not pointing it to any specific media folder when launching it here as my media folder is presently inside of WSL, but the output when run from WSL is largely the same (I also did not install ffmpeg separately on Windows yet, though I have done on WSL):

> PS C:\Users\Browner\mkv-episode-matcher> uv run python -m mkv_episode_matcher
> INFO:     Started server process [4136]
> INFO:     Waiting for application startup.
> 2026-01-17 21:59:49.177 | INFO     | mkv_episode_matcher.backend.main:startup_event:72 - Starting MKV Episode Matcher API
> 2026-01-17 21:59:49.177 | INFO     | mkv_episode_matcher.backend.main:warm_up_engine:75 - Background thread: Warming up Match Engine (loading Parakeet model)...
> INFO:     Application startup complete.
> INFO:     Uvicorn running on http://0.0.0.0:8001 (Press CTRL+C to quit)
> 2026-01-17 21:59:49.179 | INFO     | mkv_episode_matcher.core.engine:__init__:153 - Initializing ASR provider: parakeet with model: nvidia/parakeet-ctc-0.6b
> 2026-01-17 21:59:49.179 | INFO     | mkv_episode_matcher.asr_models:create_asr_model:451 - Using CTC decoder for model: nvidia/parakeet-ctc-0.6b
> 2026-01-17 21:59:49.180 | INFO     | mkv_episode_matcher.core.providers.asr:_ensure_loaded:43 - Loading ASR model: nvidia/parakeet-ctc-0.6b
> INFO:     127.0.0.1:64669 - "GET / HTTP/1.1" 404 Not Found
> [NeMo W 2026-01-17 21:59:52 nemo_logging:405] C:\Users\Browner\mkv-episode-matcher\.venv\Lib\site-packages\pydub\utils.py:170: RuntimeWarning: Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work
>       warn("Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work", RuntimeWarning)
> 
> [NeMo I 2026-01-17 21:59:55 nemo_logging:393] Tokenizer SentencePieceTokenizer initialized with 1024 tokens
> [NeMo W 2026-01-17 21:59:55 nemo_logging:405] If you intend to do training or fine-tuning, please call the ModelPT.setup_training_data() method and provide a valid configuration file to setup the train data loader.
>     Train config :
>     manifest_filepath: /disk1/NVIDIA/datasets/LibriSpeech_NeMo/librivox-train-all.json
>     sample_rate: 16000
>     batch_size: 16
>     shuffle: true
>     num_workers: 8
>     pin_memory: true
>     use_start_end_token: false
>     trim_silence: false
>     max_duration: 16.7
>     min_duration: 0.1
>     is_tarred: false
>     tarred_audio_filepaths: null
>     shuffle_n: 2048
>     bucketing_strategy: fully_randomized
>     bucketing_batch_size: null
> 
> [NeMo W 2026-01-17 21:59:55 nemo_logging:405] If you intend to do validation, please call the ModelPT.setup_validation_data() or ModelPT.setup_multiple_validation_data() method and provide a valid configuration file to setup the validation data loader(s).
>     Validation config :
>     manifest_filepath: /disk1/NVIDIA/datasets/LibriSpeech_NeMo/librivox-dev-clean.json
>     sample_rate: 16000
>     batch_size: 16
>     shuffle: false
>     use_start_end_token: false
>     num_workers: 8
>     pin_memory: true
> 
> [NeMo W 2026-01-17 21:59:55 nemo_logging:405] Please call the ModelPT.setup_test_data() or ModelPT.setup_multiple_test_data() method and provide a valid configuration file to setup the test data loader(s).
>     Test config :
>     manifest_filepath: null
>     sample_rate: 16000
>     batch_size: 16
>     shuffle: false
>     use_start_end_token: false
>     num_workers: 8
>     pin_memory: true
> 
> [NeMo I 2026-01-17 21:59:55 nemo_logging:393] PADDING: 0
> [NeMo I 2026-01-17 21:59:58 nemo_logging:393] Model EncDecCTCModelBPE was successfully restored from C:\Users\Browner\.cache\huggingface\hub\models--nvidia--parakeet-ctc-0.6b\snapshots\ad09ba1cc62743fbc9814de5d2016fca9096485a\parakeet-ctc-0.6b.nemo.
> 2026-01-17 21:59:58.120 | INFO     | mkv_episode_matcher.asr_models:load:205 - Loaded Parakeet TDT model: nvidia/parakeet-ctc-0.6b on cpu
> 2026-01-17 21:59:58.120 | SUCCESS  | mkv_episode_matcher.core.engine:__init__:160 - ASR provider loaded and ready
> 2026-01-17 21:59:58.121 | DEBUG    | mkv_episode_matcher.core.config_manager:load:32 - Config loaded from C:\Users\Browner\.mkv-episode-matcher\config.json
> 2026-01-17 21:59:58.121 | WARNING  | mkv_episode_matcher.core.providers.subtitles:_authenticate:132 - OpenSubtitles API key not configured
> 2026-01-17 21:59:58.121 | INFO     | mkv_episode_matcher.core.engine:_init_subtitle_providers:181 - OpenSubtitles provider enabled
> 2026-01-17 21:59:58.121 | INFO     | mkv_episode_matcher.backend.main:warm_up_engine:77 - Background thread: Match Engine ready!

**Comment by Jsakkos** [comment]
Try installing v1.0. I clearly broke some things in 1.1+. > Thanks for the quick answer!
> 
> I've been trying to use the CLI also, but it seems to just want to launch the UI anyway. I have not been able to get it to trigger any kind of onboarding so far.
> 
> 
> > PS C:\Users\Browner\mkv-episode-matcher> uv run python -m mkv_episode_matcher match "\\wsl.localhost\Ubuntu\home\john\media\Friends\"
> > INFO:     Started server process [10676]
> > INFO:     Waiting for application startup.
> > 2026-01-17 22:53:38.969 | INFO     | mkv_episode_matcher.backend.main:startup_event:72 - Starting MKV Episode Matcher API
> > 2026-01-17 22:53:38.970 | INFO     | mkv_episode_matcher.backend.main:warm_up_engine:75 - Background thread: Warming up Match Engine (loading Parakeet model)...
> > INFO:     Application startup complete.
> > INFO:     Uvicorn running on http://0.0.0.0:8001 (Press CTRL+C to quit)
> > 2026-01-17 22:53:38.972 | INFO     | mkv_episode_matcher.core.engine:__init__:153 - Initializing ASR provider: parakeet with model: nvidia/parakeet-ctc-0.6b
> > 2026-01-17 22:53:38.972 | INFO     | mkv_episode_matcher.asr_models:create_asr_model:451 - Using CTC decoder for model: nvidia/parakeet-ctc-0.6b
> > 2026-01-17 22:53:38.972 | INFO     | mkv_episode_matcher.core.providers.asr:_ensure_loaded:43 - Loading ASR model: nvidia/parakeet-ctc-0.6b
> > INFO:     127.0.0.1:58284 - "GET / HTTP/1.1" 404 Not Found
> > [NeMo W 2026-01-17 22:53:42 nemo_logging:405] C:\Users\Browner\mkv-episode-matcher\.venv\Lib\site-packages\pydub\utils.py:170: RuntimeWarning: Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work
> >       warn("Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work", RuntimeWarning)

**Comment by johnrix** [comment]
Hi, just to give you an update, I was able to get this running via CLI and GUI. So far, I've had mostly failures though.

On WSL:
* GUI cannot identify any drives for me to select, so dead in the water there at the moment
* On CLI, I tried it with the first 8 episodes of Season 2 of Friends. It matched 1 episode on the first pass. I adjusted the model and confidence, but was not able to get any further matches.

On Windows:
* GUI working well now 👍
* Tried matching Seasons 1 and 2. It downloaded subtitles until my API quota ran out for the day. However, there seems to be some problem reading the mkv files as I am seeing "The system cannot find the file specified" for all files in the scanned folder. Have you come across this?


2026-01-21 09:27:09.605 | INFO     | mkv_episode_matcher.core.engine:_get_subtitles_with_fallback:321 - Fetching subtitles for Friends S01
2026-01-21 09:27:09.606 | INFO     | mkv_episode_matcher.backend.routers.websocket:websocket_endpoint:18 - WebSocket disconnected
INFO:     connection closed
2026-01-21 09:27:09.607 | INFO     | mkv_episode_matcher.core.providers.subtitles:get_subtitles:377 - Found 20 cached subtitles for Friends S01
2026-01-21 09:27:09.607 | INFO     | mkv_episode_matcher.core.providers.subtitles:get_subtitles:385 - Using cached subtitles, skipping download
2026-01-21 09:27:09.607 | SUCCESS  | mkv_episode_matcher.core.engine:_get_subtitles_with_fallback:327 - Found 20 subtitle files for Friends S01
2026-01-21 09:27:09.609 | ERROR    | mkv_episode_matcher.core.utils:get_video_duration:110 - Failed to get duration for C:\Friends\Season 1\05.mkv: [WinError 2] The system cannot find the file specified
2026-01-21 09:27:09.609 | WARNING  | mkv_episode_matcher.core.matcher:match:83 - Video too short: 0.0s
2026-01-21 09:27:09.612 | ERROR    | mkv_episode_matcher.core.utils:get_video_duration:110 - Failed to get duration for C:\Friends\Season 1\06.mkv: [WinError 2] The system cannot find the file specified
2026-01-21 09:27:09.612 | WARNING  | mkv_episode_matcher.core.matcher:match:83 - Video too short: 0.0s
